src/main/java/org/zone/commands/structure/join/JoinZoneCommand.java

```diff
@@ -59,9 +60,13 @@
                 .getFlag(FlagTypes.ZONE_VISIBILITY)
                 .map(ZoneVisibilityFlag::getZoneVisibility)
                 .orElse(ZoneVisibility.PUBLIC);
+        BanFlag banFlag = new BanFlag();
```

Comment:
Did you mean to get the BanFlag from the zone before creating a new one?

Reply:
Oh yeah XD